### PR TITLE
Add support for `HashMap`, `HashSet`

### PR DESCRIPTION
**Pull request by tbu-**

### CODE REVIEW

**Comment by tbu-** [comment]
Depends on https://github.com/japaric/syscall.rs/pull/9.

**Comment by tbu-** [comment]
CC #5.

**Comment by japaric** [comment]
Why do we need to copy the `rand` crate wholesale? Can we just use the one provided by the `rust-src` component, i.e. by adding it as a dependency in Xargo.toml?

File: src/collections/hash/map.rs [hunk 1]
```diff
@@ -2054,17 +2054,15 @@ impl RandomState {
         // iteration order allows a form of DOS attack. To counter that we
         // increment one of the seeds on every RandomState creation, giving
         // every corresponding HashMap a different iteration order.
-        thread_local!(static KEYS: Cell<(u64, u64)> = {
```
Comment (japaric) [inline src/collections/hash/map.rs]: If we are not going to use TLS here, I'd rather keep the TLS version commented out and just use [the previous implementation](https://github.com/rust-lang/rust/pull/33318/files#diff-1ab410241f658d595f1c5eec7a979b01L1670).

Comment (japaric) [reply src/collections/hash/map.rs]: Also, could you add a note here and the tracking issue that we still have to move this to use TLS?

**Comment by tbu-** [comment]
@homunkulus try

**Comment by homunkulus** [comment]
:hourglass: Trying commit 025b6e6 with merge 025b6e6...

**Comment by homunkulus** [comment]
:broken_heart: Test failed - [status-travis](https://travis-ci.org/japaric/steed/builds/197604335)

**Comment by tbu-** [comment]
@homunkulus try

**Comment by homunkulus** [comment]
:hourglass: Trying commit 8a8a997 with merge 8a8a997...

**Comment by homunkulus** [comment]
:broken_heart: Test failed - [status-travis](https://travis-ci.org/japaric/steed/builds/197614261)

**Comment by tbu-** [comment]
@japaric I had the same error locally, it was fixed by deleting the `target` directory.

**Comment by japaric** [comment]
> @japaric I had the same error locally, it was fixed by deleting the target directory.

I have blown up the `target` caches.

@tbu-, could you squash the first two commits so we don't end with two huge diffs in the git history? Also, could you move to the latest stable release of the `sc` crate? It contains the GETRANDOM syscall. Thanks.

**Comment by tbu-** [comment]
@homunkulus try

**Comment by homunkulus** [comment]
:hourglass: Trying commit eb560ac with merge eb560ac...

**Comment by homunkulus** [comment]
:broken_heart: Test failed - [status-travis](https://travis-ci.org/japaric/steed/builds/197843114)

**Comment by tbu-** [comment]
Hmmm.

**Comment by tbu-** [comment]
Now depends on https://github.com/japaric/syscall.rs/pull/10.

**Comment by tbu-** [comment]
@homunkulus retry

**Comment by homunkulus** [comment]
:hourglass: Trying commit eb560ac with merge 0300411...

**Comment by tbu-** [comment]
@homunkulus abort

**Comment by tbu-** [comment]
@homunkulus try

**Comment by homunkulus** [comment]
:hourglass: Trying commit 5b23f0d with merge 2fa8db7...

**Comment by homunkulus** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/japaric/steed/builds/198101208)
State: approved= try=True

**Comment by japaric** [comment]
Thanks, @tbu-!

**Pull request merged**